src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -189,6 +189,43 @@ class ContentSidebar extends PureComponent<Props, State> {
         onInteraction(Object.assign({}, { file: cloneDeep(file) }, data));
     };
 
+    /**
+     * Function to update file description
+     *
+     * @private
+     * @param {string} value - new file description
+     * @return {void}
+     */
+    onDescriptionChange = (value: string): void => {
+        const { file } = this.state;
+        if (!file) {
+            return;
+        }
+
+        const { description, id } = file;
+        if (value === description || !description || !id) {
+            return;
+        }
+
+        this.api.getFileAPI().setFileDescription(id, value, this.setFileDescriptionCallback, this.errorCallback);
+    };
+
+    /**
+     * File update description callback
+     *
+     * @private
+     * @param {string} value - updated file description
+     * @return {void}
+     */
+    setFileDescriptionCallback = (value: string): void => {
+        this.setState({
```

Comment:
This needs to be done at the API level, the file object in cache should be updated.

Reply:
I have started adding event like stuff for analytics, look at onInteraction
eg: https://github.com/box/box-ui-elements/blob/473ae7679480c4e973b754fc3b3e60b68ee710d6/src/components/Transcript/TranscriptDialog.js#L60

Probably add one for description change.